src/Client.php

```diff
@@ -176,7 +176,12 @@ private function prepareEvent(array $payload, ?Scope $scope = null, bool $withSt
             return null;
         }
 
-        if ($withStacktrace) {
+        // when 'exception' is set, no backtrace should be auto-attached (#957)
```

Comment:
I can do that. The reason I put the comment there is because back in #940 where an explicit test for this was removed, there was a comment posted where people didn't remember what the check was for.

As this behavior is unexpected and has been broken once even though it was there previously, I felt like it warranted the comment.

Reply:
We should be covered by documenting each test case of this piece of code: in the `captureEventAttachesStacktraceAccordingToAttachStacktraceOptionDataProvider` method we should return a custom description for each test case similarly to what has been done in `addFrameSetsInAppFlagCorrectlyDataProvider` so that if for some reason we change this code and we break an existing test case it's clear why we did that thing in the past

Reply:
fixed.